### PR TITLE
Chunk large reads into multiple smaller reads of max 64MB.

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge PR

### CODE REVIEW

File: fastos/src/vespa/fastos/linux_file.cpp [hunk 1]
```diff
@@ -44,7 +44,18 @@ FastOS_Linux_File::readInternal(int fh, void *buffer, size_t length, int64_t rea
 ssize_t
 FastOS_Linux_File::readInternal(int fh, void *buffer, size_t length)
 {
-    return File_RW_Ops::read(fh, buffer, length);
+    char * data = static_cast<char *>(buffer);
+    ssize_t read(0);
```
Comment (toregge) [inline fastos/src/vespa/fastos/linux_file.cpp:48]: Consider a different name for the variable, e.g. `has_read`

**Review by toregge — APPROVED** [review]

File: fastos/src/vespa/fastos/linux_file.cpp [hunk 2]
```diff
@@ -37,7 +37,18 @@ FastOS_Linux_File::FastOS_Linux_File(const char *filename)
 ssize_t
 FastOS_Linux_File::readInternal(int fh, void *buffer, size_t length, int64_t readOffset)
 {
-    return File_RW_Ops::pread(fh, buffer, length, readOffset);
+    char * data = static_cast<char *>(buffer);
+    ssize_t read(0);
```
Comment (toregge) [inline fastos/src/vespa/fastos/linux_file.cpp:41]: Consider a different name for the variable, e.g. `has_read`

**Review by toregge — APPROVED** [review]

File: fastos/src/vespa/fastos/linux_file.cpp (hunk 1 above)
Comment (baldersheim) [reply fastos/src/vespa/fastos/linux_file.cpp]: Fixed

File: fastos/src/vespa/fastos/linux_file.cpp (hunk 2 above)
Comment (baldersheim) [reply fastos/src/vespa/fastos/linux_file.cpp]: done

**Review by toregge — APPROVED** [review]

**Pull request merged**